include/crocoddyl/core/optctrl/shooting.hxx

```diff
@@ -317,7 +317,7 @@ void ShootingProblemTpl<Scalar>::circularAppend(boost::shared_ptr<ActionModelAbs
 template <typename Scalar>
 void ShootingProblemTpl<Scalar>::updateNode(std::size_t i, boost::shared_ptr<ActionModelAbstract> model,
                                             boost::shared_ptr<ActionDataAbstract> data) {
-  if (i > T_ + 1) {
+  if ((i >= T_ + 1) || (i < 0)) {
```

Comment:
There is a minimal detail here. `std::size_t` is an unsigned integer, therefore, we don't need to include `|| (i < 0)`.

Reply:
DOne.